crypto/encrypt-decrypt.go

```diff
@@ -50,35 +51,64 @@ func Base64MD5Hash(byteStream []byte) string {
 	return base64MD5Hash
 }
 
+func GetKMSEnvelopeClient(resourceName string) (*aead.KMSEnvelopeAEAD, error){
```

Comment:
Looks good, please rename 
GetKMSEnvelopeClient => getCachedKMSEnvelopeClient 
and
getNewKMSEnvelopeClient => getKMSEnvelopeClient

This will aid readability.

Reply:
OR maybe better,
chage  GetKMSEnvelopeClient to return getNewKMSEnvelopeClient if not found instead of an error, and onluy call GetKMSEnvelopeClient

I kinda like that a little better but up to you.